### PR TITLE
Cleanup pom, add bom

### PR DESCRIPTION
**Pull request by timja**
@KostyaSha

if this is merged then https://github.com/jenkinsci/github-plugin/pull/237 can be closed

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -50,15 +50,18 @@
         <revision>1.33.0</revision>
         <changelist>-SNAPSHOT</changelist>
         <gitHubRepo>jenkinsci/github-plugin</gitHubRepo>
-        <jenkins.version>2.164.3</jenkins.version>
+        <jenkins.version>2.222.4</jenkins.version>
```
Comment (timja) [inline pom.xml:53]: 97% of users are on this version (https://stats.jenkins.io/pluginversions/github.html), and it's the minimum supported for new versions of bom currently.

also see https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/

File: src/test/java/org/jenkinsci/plugins/github/status/sources/ManuallyEnteredRepositorySourceTest.java [hunk 2]
```diff
@@ -33,12 +32,11 @@
 
     @Test
     public void nullName() {
-        ManuallyEnteredRepositorySource instance = Mockito.spy(new ManuallyEnteredRepositorySource("https://github.com/jenkinsci/jenkins"));
-        doReturn(null).when(instance).createName(Matchers.anyString());
+        ManuallyEnteredRepositorySource instance = spy(new ManuallyEnteredRepositorySource("a"));
```
Comment (timja) [inline src/test/java/org/jenkinsci/plugins/github/status/sources/ManuallyEnteredRepositorySourceTest.java:35]: some weird mockito issue that only occurred when running all tests via CLI, couldn't reproduce in IDE or `mvn test`.

the test is checking nullName, so faking an impossible case didn't make sense anymore so I removed the mocking and the test works

File: src/test/resources/checkstyle/checkstyle-config.xml [hunk 3]
```diff
@@ -185,10 +185,6 @@
         <!-- Miscellaneous other checks.                   -->
         <!-- See http://checkstyle.sf.net/config_misc.html -->
         <module name="ArrayTypeStyle"/>
-        <module name="TodoComment">
```
Comment (timja) [inline src/test/resources/checkstyle/checkstyle-config.xml:188]: this is what is causing other PRs to fail CI currently

File: src/test/resources/checkstyle/checkstyle-config.xml [hunk 4]
```diff
@@ -32,6 +32,7 @@
 
 
 <module name="Checker">
+    <property name="cacheFile" value="${checkstyle.cache.file}"/>
```
Comment (timja) [inline src/test/resources/checkstyle/checkstyle-config.xml:35]: most changes in this file are just moving them around due to checkstyle update

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**